src/appleseed/renderer/modeling/camera/camera.cpp

```diff
@@ -452,6 +343,115 @@ void Camera::check_shutter_times_for_consistency() const
     }
 }
 
+void Camera::initialize_shutter_curve_linear()
+{
+    float cbt_minus_oet_plus_one =
+        impl->m_normalized_close_begin_time - impl->m_normalized_open_end_time + 1.0f;
+
+    if (cbt_minus_oet_plus_one == 0.0f)
+    {
+        RENDERER_LOG_ERROR(
+            "while defining camera \"%s\": invalid values \"%f\" \"%f\" for parameters \"%s\", \"%s\" "
+            "using default values for parameters \"%s\", \"%s\", \"%s\", \"%s\" "
+            "\"%f\", \"%f\", \"%f\", \"%f\".",
+            get_path().c_str(),
+            m_shutter_open_end_time,
+            m_shutter_close_begin_time,
+            "shutter_open_end_time",
+            "shutter_close_begin_time",
+            "shutter_open_begin_time",
+            "shutter_open_end_time",
+            "shutter_close_begin_time",
+            "shutter_close_begin_time",
+            ShutterOpenBeginTimeDefaultValue,
+            ShutterOpenBeginTimeDefaultValue,
+            ShutterCloseEndTimeDefaultValue,
+            ShutterCloseEndTimeDefaultValue);
+
+        m_shutter_open_begin_time = ShutterOpenBeginTimeDefaultValue;
+        m_shutter_open_end_time = ShutterOpenBeginTimeDefaultValue;
+        m_shutter_close_begin_time = ShutterCloseEndTimeDefaultValue;
+        m_shutter_close_end_time = ShutterCloseEndTimeDefaultValue;
+
+        impl->m_normalized_open_end_time =
+            inverse_lerp(
+                m_shutter_open_begin_time,
+                m_shutter_close_end_time,
+                m_shutter_open_end_time);
+
+        impl->m_normalized_close_begin_time =
+            inverse_lerp(
+                m_shutter_open_begin_time,
+                m_shutter_close_end_time,
+                m_shutter_close_begin_time);
+
+        m_shutter_time_interval = m_shutter_close_end_time - m_shutter_open_begin_time;
+
+        cbt_minus_oet_plus_one =
+            impl->m_normalized_close_begin_time - impl->m_normalized_open_end_time + 1.0f;
+    }
+
+    impl->m_shutter_curve_linear_open_multiplier =
+        cbt_minus_oet_plus_one * impl->m_normalized_open_end_time;
+
+    impl->m_shutter_curve_linear_fully_opened_multiplier = cbt_minus_oet_plus_one / 2.0f;
```

Comment:
Better to multiply by 0.5 than do a quotient

Reply:
True. Note that this is called once before rendering starts, not during rendering.